src/main/java/address/sync/CloudService.java

```diff
@@ -0,0 +1,394 @@
+        }
+        return new ExtractedCloudResponse<>(cloudResponse.getResponseCode(), getRateLimitFromHeader(headerHashMap),
+                                            getRateRemainingFromHeader(headerHashMap),
+                                            getRateResetFromHeader(headerHashMap), null);
+    }
+
+    private <V> ExtractedCloudResponse<V> getResponseWithNoData(RawCloudResponse cloudResponse,
+                                                                HashMap<String, Long> headerHashMap) {
+        return new ExtractedCloudResponse<>(cloudResponse.getResponseCode(), getRateLimitFromHeader(headerHashMap),
+                                            getRateRemainingFromHeader(headerHashMap),
+                                            getRateResetFromHeader(headerHashMap), null);
+    }
+
+    private BufferedReader getReaderForStream(InputStream stream) {
+        return new BufferedReader(new InputStreamReader(stream));
+    }
+
+    private String convertToString(InputStream stream) throws IOException {
+        BufferedReader reader = getReaderForStream(stream);
+        StringBuilder stringBuffer = new StringBuilder();
+        while (reader.ready()) {
+            stringBuffer.append(reader.readLine());
+        }
+
+        return stringBuffer.toString();
+    }
+
+    /**
+     * Parses the stream content and attempts to convert it into an object T
+     * @param bodyStream
+     * @param type
+     * @param <T>
+     * @return
+     * @throws IOException
+     */
+    private <T> T getDataFromBody(InputStream bodyStream, Class<T> type) throws IOException {
+        return JsonUtil.fromJsonString(convertToString(bodyStream), type);
+    }
+
+    /**
+     * Parses the stream content and attempts to convert it into a List of object Ts
+     *
+     * @param bodyStream
+     * @param type
+     * @param <T>
+     * @return
+     * @throws IOException
+     */
+    private <T> List<T> getDataListFromBody(InputStream bodyStream, Class<T> type) throws IOException {
+        return JsonUtil.fromJsonStringToList(convertToString(bodyStream), type);
+    }
+
+    /**
+     * Parses the JSON-formatted stream and attempts to get the header in HashMap<String, Long> form
+     *
+     * @param headerStream
+     * @return
+     * @throws IOException
+     */
+    private HashMap<String, Long> getHashMapFromHeader(InputStream headerStream) throws IOException {
+        return JsonUtil.fromJsonStringToHashMap(
+                convertToString(headerStream), String.class, Long.class);
+    }
+
+    private int getRateLimitFromHeader(HashMap<String, Long> header) {
+        return header.get("X-RateLimit-Limit").intValue();
+    }
+
+    private int getRateRemainingFromHeader(HashMap<String, Long> header) {
+        return header.get("X-RateLimit-Remaining").intValue();
+    }
+
+    private long getRateResetFromHeader(HashMap<String, Long> header) {
+        return header.get("X-RateLimit-Limit");
```

Comment:
Is this correct key? Looks the same with `getRateLimitFromHeader` except not converted to `intValue()`. Maybe if that's the case, `getRateLimitFromHeader` should call this method in which it will call `intValue()`.